framework/src/play/src/main/scala/play/api/ApplicationLoader.scala

```
@@ -53,13 +56,82 @@ object ApplicationLoader {
+    sourceMapper: SourceMapper,
+    buildLink: BuildLink
+  )
+
+  object Context {
+
+    /**
+     * Create an application loading context.
+     *
+     * Locates and loads the necessary configuration files for the application.
+     *
+     * @param environment The application environment.
+     * @param initialSettings The initial settings. These settings are merged with the settings from the loaded
+     *                        configuration files, and together form the initialConfiguration provided by the context.  It
+     *                        is intended for use in dev mode, to allow the build system to pass additional configuration
+     *                        into the application.
+     * @param lifecycle Used to register hooks that run when the application stops.
+     * @param devContext If an application is loaded in dev mode then this additional context can be provided.
+     */
+    def create(
+      environment: Environment,
+      initialSettings: Map[String, AnyRef] = Map.empty[String, AnyRef],
+      lifecycle: ApplicationLifecycle = new DefaultApplicationLifecycle(),
+      devContext: Option[DevContext] = None) = {
+      Context(
+        environment = environment,
+        devContext = devContext,
+        lifecycle = lifecycle,
+        initialConfiguration = Configuration.load(environment, initialSettings)
+      )
+    }
+
+    @deprecated("Context properties have changed; use the default Context apply method or Context.create instead", "2.7.0")
+    def apply(
+      environment: Environment,
+      sourceMapper: Option[SourceMapper],
+      webCommands: WebCommands,
+      initialConfiguration: Configuration,
+      lifecycle: ApplicationLifecycle): Context = {
+      require(sourceMapper == None, "sourceMapper parameter is no longer supported by ApplicationLoader.Context; use devContext parameter instead")
+      require(webCommands == null, "webCommands parameter is no longer supported by ApplicationLoader.Context")
```

This is disruptive - but better to fail early, right?